util/fpga/splice_nexysvideo.sh

```diff
@@ -30,18 +30,16 @@ FPGA_BIT_NAME=lowrisc_systems_chip_earlgrey_nexysvideo_0.1
 ./meson_init.sh
 ninja -C "${OBJ_DIR}" "${TARGET_EXPORT}"
 
-srec_cat "${TARGET}.bin" -binary -offset 0x0 -o "${TARGET}.brammem" \
-  -vmem -Output_Block_Size 4;
-
-util/fpga/addr4x.py -i "${TARGET}.brammem" -o "${TARGET}.mem"
+# Create the Vivado image for splicing.
+hw/ip/rom_ctrl/util/gen_vivado_mem_image.py "${TARGET}.32.vmem" "${TARGET}_vivado.mem"
```

Comment:
mem and vmem files are effectively different variations of the same file format; the file extension in itself isn't really describing it. (vmem files as we use them are also given a .mem suffix from time to time). Also, Vivado uses different variations of the file format in different places, with updatemem having its own requirements. (https://www.xilinx.com/support/documentation/sw_manuals/xilinx2020_2/ug898-vivado-embedded-design.pdf#page=165 for the beautiful details).

To reduce confusion, would you mind renaming this file to something like `"${TARGET}.updatemem.mem"` (also keeping the use of a dot for further qualifying the file, as we do for `.scr.` etc.), and update the comment in the gen_vivado_mem_image.py to better describe this?

Reply:
Thanks @imphil for providing some more insight here. I've added some more details + the ref to `gen_vivado_mem_image.py` and renamed the files as suggested.